src/main/java/nextstep/subway/favorite/domain/FavoriteRepository.java

```diff
@@ -0,0 +1,19 @@
+package nextstep.subway.favorite.domain;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import nextstep.subway.member.domain.Member;
+
+public interface FavoriteRepository extends JpaRepository<Favorite, Long> {
+	@Query("select f "
+		+ "from Favorite f "
+		+ "join fetch f.creator "
+		+ "join fetch f.target "
+		+ "join fetch f.source "
+		+ "where f.creator = :creator")
+	List<Favorite> findAllWithStationByCreator(@Param("creator") Member creator);
```

Comment:
`@Query` 사용 잘해주셨습니다 👍 
추가적으로 `Spring Jpa`에서 제공하는 `Query creation`(자동생성) 기능이 있습니다 😄 
이미 `Favorite`에 `creator`를 정의하셨으므로 `ByCreator`를 통해 동일한 결과를 조회할 수 있습니다 😄 

이 부분은 참고만 부탁 드립니다 🙇  
```suggestion
	List<Favorite> findAllByCreator(Member creator);
```
> https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.query-creation

Reply:
네 링크까지ㅜ 감사합니다!
다만 제가 쿼리를 따로 작성한 이유는 현재 Favorite 에서 Station 의 FetchType 을 LAZY 로 잡아놨습니다. (`N + 1 문제로 인해 many to one 관계에서는 fetchType 을 LAZY 로 잡고, 필요하면 fetch join 이나 entity graph 등을 통해 최적화 하는게 좋다.` 라는 말을 어느 강의에서 들었기 때문입니다.)

때문에  `List<Favorite> findAllByCreator(Member creator);` 메서드를 실행하면,  `sourceStation`,  `targetStation` 을 가지고 오기 위해 쿼리를 따로 더 날리더라구요!

저는 fetch join 을 학습하고 싶기도 하고 `Favorite`, `Creator`, `sourceStation`,  `targetStation` 모두 한쿼리에 가지고 오면 더 좋겠다고 생각이 들어 따로 쿼리를 작성하게 되었습니다.